CoreFoundation/RunLoop.subproj/CFRunLoop.c

```diff
@@ -1306,7 +1306,7 @@ static Boolean __CFRunLoopSourceIsSignaled(CFRunLoopSourceRef rls) {
 }
 
 CF_INLINE void __CFRunLoopSourceSetSignaled(CFRunLoopSourceRef rls) {
-    atomic_compare_exchange_strong_explicit(&rls->_signaledTime, &(uint64_t){0}, mach_absolute_time(), memory_order_acq_rel, memory_order_acq_rel);
+    atomic_compare_exchange_strong_explicit(&rls->_signaledTime, &(uint64_t){0}, mach_absolute_time(), memory_order_acq_rel, memory_order_acquire);
```

Comment:
Actually, because (uint64_t){0} is an array that decays to a pointer to an array that isn't used later, and mach_absolute_time() is not dependent on the value of &rls->_signaledTime, we can replace the success and failure order argument with memory_order_release, and memory_order_relaxed respectively.

Reply:
Addendum:

```C
CF_INLINE void __CFRunLoopSourceSetSignaled(CFRunLoopSourceRef rls) {
    atomic_load_explicit(&rls->_signaledTime, mach_absolute_time(), memory_order_release);
}
```
Would actually be the ideal way